packages/jest-reporters/src/GithubActionsReporter.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {GITHUB_ACTIONS} from 'ci-info';
+import type {AggregatedResult, TestResult} from '@jest/test-result';
+import BaseReporter from './BaseReporter';
+import type {Context} from './types';
+
+const lineAndColumnInStackTrace = /^.*?:([0-9]+):([0-9]+).*$/;
+
+function replaceEntities(s: string): string {
+  const substitutions: Array<[RegExp, string]> = [
+    [/%/, '%25'],
+    [/\r/g, '%0D'],
+    [/\n/g, '%0A'],
+  ];
+  return substitutions.reduce((acc, sub) => acc.replace(...sub), s);
+}
+
+export default class GithubActionsReporter extends BaseReporter {
+  onRunComplete(
+    _contexts?: Set<Context>,
+    aggregatedResults?: AggregatedResult,
+  ): void {
+    if (!GITHUB_ACTIONS) {
```

Comment:
question - should the reporter care? I'm thinking it makes more sense for the reporter to print when it's active, regardless of env. Then it's up to some other part of the code whether to activate it or not

Reply:
sure, but that's separate. We only enable coverage reporter of coverage is active (https://github.com/facebook/jest/blob/a20bd2c31e126fc998c2407cfc6c1ecf39ead709/packages/jest-core/src/TestScheduler.ts#L350-L381) we should do the same for a GH reporter. The reporter itself should always print if it's active, and the env detection should deicide to enable the reporter or not

Reply:
Gotcha! I'll remove the GHA check altogether then 👍

Reply:
For the defaults, should I add something like this to the PR?

<details>
<summary>
Patch
</summary>

```diff
diff --git a/packages/jest-core/src/TestScheduler.ts b/packages/jest-core/src/TestScheduler.ts
index 17fdff1935..7a9397e4e3 100644
--- a/packages/jest-core/src/TestScheduler.ts
+++ b/packages/jest-core/src/TestScheduler.ts
@@ -8,10 +8,12 @@
 /* eslint-disable local/ban-types-eventually */
 
 import chalk = require('chalk');
+import {GITHUB_ACTIONS} from 'ci-info';
 import exit = require('exit');
 import {
   CoverageReporter,
   DefaultReporter,
+  GitHubActionsReporter,
   NotifyReporter,
   Reporter,
   SummaryReporter,
@@ -347,11 +349,15 @@ class TestScheduler {
   }
 
   async _setupReporters() {
-    const {collectCoverage, notify, reporters} = this._globalConfig;
+    const {annotateGHA, collectCoverage, notify, reporters} =
+      this._globalConfig;
     const isDefault = this._shouldAddDefaultReporters(reporters);
 
     if (isDefault) {
-      this._setupDefaultReporters(collectCoverage);
+      this._setupDefaultReporters(
+        collectCoverage,
+        annotateGHA !== false && GITHUB_ACTIONS,
+      );
     }
 
     if (!isDefault && collectCoverage) {
@@ -364,6 +370,10 @@ class TestScheduler {
       );
     }
 
+    if (!isDefault && annotateGHA) {
+      this.addReporter(new GitHubActionsReporter());
+    }
+
     if (notify) {
       this.addReporter(
         new NotifyReporter(
@@ -379,7 +389,10 @@ class TestScheduler {
     }
   }
 
-  private _setupDefaultReporters(collectCoverage: boolean) {
+  private _setupDefaultReporters(
+    collectCoverage: boolean,
+    annotateGHA: boolean,
+  ) {
     this.addReporter(
       this._globalConfig.verbose
         ? new VerboseReporter(this._globalConfig)
@@ -396,6 +409,10 @@ class TestScheduler {
       );
     }
 
+    if (annotateGHA) {
+      this.addReporter(new GitHubActionsReporter());
+    }
+
     this.addReporter(new SummaryReporter(this._globalConfig));
   }

```
</details>

Or should that be done in a follow-up?

Reply:
let's do that in a follow-up!

Reply:
Feel free to open up a PR when ready, btw. My main concern is if somebody wants to turn it _off_, that might be weird. But let's see! 🙂 

Regardless, we should add `'github-actions'` as a recognized string in the `reporters` array (like `default` is)